README.md

```diff
@@ -1,9 +1,310 @@
+<!-- TOC -->
 # osde2e-example-test-harness
 
-This is an example test harness meant for testing the osde2e example operator addon. It does the following:
+Test harnesses are standalone ginkgo test images run on test pods on test clusters by osde2e framework in prow jobs. 
 
-* Tests for the existence of the prowjobs.prow.k8s.io CRD. This should be present if the prow
-  operator addon has been installed properly.
-* Writes out a junit XML file with tests results to the /test-run-results directory as expected
-  by the [https://github.com/openshift/osde2e](osde2e) test framework.
-* Writes out an `addon-metadata.json` file which will also be consumed by the osde2e test framework.
+This respository is an example of a test harness.  It uses the Reference Addon as an example and demonstrates basic test assertions. 
+It does the following:
+
+* Contains source for example test-harness image published in: (quay.io/rmundhe_oc/osde2e-example-test-harness)
+* Asserts basic funcitonality of the addon and the test harness. e.g. existence of _reference-addon_ CRD  *referenceaddons.reference.addons.managed.openshift.io*.  
+
+When `osde2e` test framework executes this harness, it writes out 
+1. a junit XML file with tests results to the `/test-run-results` directory to reflect test results, and
+2. `addon-metadata.json` file to be consumed by the osde2e framework
+
+ 
+This doc explains how to execute these tests locally as well as how to create prow jobs to schedule a periodic pipeline. You may use this example to create your own addon test harness and prow jobs.
+
+> The addons integration (e2e) tests are not meant to replace any existing QE.
+ This document is not a reference for complete onboarding procedure for addons to OSD. Full process of onboarding addons is outlined in the documentation [here](https://gitlab.cee.redhat.com/service/managed-tenants/-/tree/master).
+ 
+
+## The Structure of an Addon Test
+
+How an add-on is tested can vary between groups and projects. In light of this, we've tried to create a very flexible and unopinionated framework for your testing. Your test harness should take the form of an OCI (docker) container that does the following:
+
+*   Assume it is executing in a pod within an OpenShift cluster.
+*   Assume the pod will inherit `cluster-admin` rights.
+*	Block until your addon is ready to be tested (we will launch your container after requesting installation of the addon, but we can't control when the addon is finished installing).
+*   Output a valid `junit.xml` file to the `/test-run-results` directory before the container exits.
+*   Output metadata to `addon-metadata.json` in the `/test-run-results` directory.
+
+
+## Locally Running This Example 
+
+1. Create a stage rosa cluster
+2. Install your addon/s on it
+3. Clone osde2e: `git clone git@github.com:openshift/osde2e.git`
+4. Build osde2e executable: `make build`
+5. Run osde2e as follows
+
+  ```bash
```

Comment:
This local config assumes a they are passing a cluster in an already state. 
I wouldn't say these flags are all needed for a local run, but it's a common case. 

I'd spell out that things config works with a long running cluster and for more power user availability to look at the osde2e's config.md

Reply:
I've written this section for SOP for a dev trying to  run an example, so everything you need for local testing is included here. It's save some headache to get the example running - for our future teammates.  
However, other configs are described in the Parameters section. They will need to read the whole doc. Is that sufficient or do we need a change here?

Reply:
If it helps, the detailed config section is a bit further down. https://github.com/ritmun/osde2e-example-test-harness/blob/sdcicd-834-docs-update/README.md